doc/trajopt_planner/src/test_trajopt.cpp

```diff
@@ -0,0 +1,269 @@
+  req.goal_constraints.clear();
+  req.group_name = PLANNING_GROUP;
+  // set the middle and goal state and joints tolerance
+  // ========================================================================================
+  robot_state::RobotStatePtr middle_state(new robot_state::RobotState(robot_model));
+  std::vector<double> middle_joint_values = { 0.5, 0.4, 0.65, -0.75, 1.05, 1.25, -0.15 };
+  middle_state->setJointGroupPositions(joint_model_group, middle_joint_values);
+  middle_state->update();
+  moveit_msgs::Constraints joint_middle =
+      kinematic_constraints::constructGoalConstraints(*middle_state, joint_model_group);
+  req.goal_constraints.push_back(joint_middle);
+  req.goal_constraints[0].name = "middle_pos";
+  // set joint tolerance
+  std::vector<moveit_msgs::JointConstraint> middle_joint_constraint = req.goal_constraints[0].joint_constraints;
+  for (int x = 0; x < middle_joint_constraint.size(); ++x)
+  {
+    ROS_INFO(" ======================================= joint position at goal: %f",
+             middle_joint_constraint[x].position);
+    req.goal_constraints[0].joint_constraints[x].tolerance_above = 0.001;
+    req.goal_constraints[0].joint_constraints[x].tolerance_below = 0.001;
+  }
+
+  robot_state::RobotStatePtr goal_state(new robot_state::RobotState(robot_model));
+  std::vector<double> goal_joint_values = { 0.8, 0.7, 1, -1.3, 1.9, 2.2, -0.1 };
+  goal_state->setJointGroupPositions(joint_model_group, goal_joint_values);
+  goal_state->update();
+  moveit_msgs::Constraints joint_goal = kinematic_constraints::constructGoalConstraints(*goal_state, joint_model_group);
+  req.goal_constraints.push_back(joint_goal);
+  req.goal_constraints[1].name = "goal_pos";
+  // set joint tolerance
+  std::vector<moveit_msgs::JointConstraint> goal_joint_constraint = req.goal_constraints[1].joint_constraints;
+  for (int x = 0; x < goal_joint_constraint.size(); ++x)
+  {
+    ROS_INFO(" ======================================= joint position at goal: %f", goal_joint_constraint[x].position);
+    req.goal_constraints[1].joint_constraints[x].tolerance_above = 0.001;
+    req.goal_constraints[1].joint_constraints[x].tolerance_below = 0.001;
+  }
+
+  // Load planner
+  // ======================================================================================
+  while (!psm->getStateMonitor()->haveCompleteState() && ros::ok())
+  {
+    ROS_INFO_STREAM_THROTTLE_NAMED(1, node_name, "Waiting for complete state from topic ");
+  }
+  // We will now construct a loader to load a planner, by name.
+  // Note that we are using the ROS pluginlib library here.
+  boost::scoped_ptr<pluginlib::ClassLoader<planning_interface::PlannerManager>> planner_plugin_loader;
+  planning_interface::PlannerManagerPtr planner_instance;
+
+  std::string planner_plugin_name;
+
+  planner_plugin_name = "trajopt_interface/TrajOptPlanner";
+  node_handle.setParam("planning_plugin", planner_plugin_name);
+
+  // making sure to catch all exceptions.
```

Comment:
not sure what this comment means

Reply:
"Make sure the planner plugin is loaded" instead?